doc/tutorial/text_analytics/data/twenty_newsgroups/fetch_data.py

```diff
@@ -21,9 +21,16 @@
     if not os.path.exists(ARCHIVE_NAME):
         print("Downloading dataset from %s (14 MB)" % URL)
         opener = urlopen(URL)
-        open(ARCHIVE_NAME, 'wb').write(opener.read())
+        with open(ARCHIVE_NAME,'wb') as archive:
+            archive.write(opener.read())
 
     print("Decompressing %s" % ARCHIVE_NAME)
-    tarfile.open(ARCHIVE_NAME, "r:gz").extractall(path='.')
+    try:
+        with tarfile.open(ARCHIVE_NAME, "r:gz") as archive:
+            archive.extractall(path = '.')
+    except AttributeError: # In Python 2.6, tarfile did not yet implement the context manager protocol
```

Comment:
Instead of the try / except construct to preserver Python 2.6 compat I think you can use the the `closing` contextmanager:

```python
from contextlib import closing

with closing(tarfile.open(ARCHIVE_NAME, "r:gz")) as archive:
    archive.extractall(path = '.')
```